lib/ansible/modules/database/postgresql/postgresql_set.py

```diff
@@ -0,0 +1,518 @@
+        raw_val = int(raw_val) * 1024
+        boot_val = int(boot_val) * 1024
+        unit = 'b'
+
+    elif unit == 'MB':
+        raw_val = int(raw_val) * 1024 * 1024
+        boot_val = int(boot_val) * 1024 * 1024
+        unit = 'b'
+
+    return (val[0], raw_val, unit, boot_val, context)
+
+
+def pretty_to_bytes(pretty_val):
+    # The function returns a value in bytes
+    # if the value contains 'B', 'kB', 'MB', 'GB', 'TB'.
+    # Otherwise it returns the passed argument.
+
+    val_in_bytes = None
+
+    if 'kB' in pretty_val:
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024
+
+    elif 'MB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024
+
+    elif 'GB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024 * 1024
+
+    elif 'TB' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part * 1024 * 1024 * 1024 * 1024
+
+    elif 'B' in pretty_val.upper():
+        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
+        val_in_bytes = num_part
+
+    else:
+        return pretty_val
+
+    return val_in_bytes
+
+
+def param_set(cursor, module, name, value):
+    try:
+        if value in ('DEFAULT', 'default'):
+            query = "ALTER SYSTEM SET %s = DEFAULT" % name
+        else:
+            query = "ALTER SYSTEM SET %s = '%s'" % (name, value)
+        cursor.execute(query)
+        cursor.execute("SELECT pg_reload_conf()")
+    except Exception as e:
+        module.fail_json(msg="Unable to get %s value due to : %s" % (name, to_native(e)))
+
+    return True
+
+
+def connect_to_db(module, kw, autocommit=False):
+    try:
+        db_connection = psycopg2.connect(**kw)
+        if autocommit:
+            db_connection.set_session(autocommit=True)
+
+    except TypeError as e:
+        if 'sslrootcert' in e.args[0]:
+            module.fail_json(msg='Postgresql server must be at least version 8.4 to support sslrootcert')
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e))
+    except Exception as e:
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e))
+
+    return db_connection
+
+# ===========================================
+# Module execution.
+#
+
+
+def main():
+    argument_spec = postgres_common_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', required=True),
+        db=dict(type='str', aliases=['login_db']),
+        port=dict(type='int', default=5432, aliases=['login_port']),
+        ssl_mode=dict(type='str', default='prefer', choices=['allow', 'disable', 'prefer', 'require', 'verify-ca', 'verify-full']),
+        ssl_rootcert=dict(type='str'),
+        value=dict(type='str'),
+        reset=dict(type='bool'),
```

Comment:
missing default value?

Reply:
there's no default value. We need to absolutely understand what we are going to set